Bug:

<!--
  Thanks for taking the time to file a bug report! Please fill this template out as completely as possible.
-->

### Checklist

- [x] Verified whether your issue exists in the latest version Mux Element(s) release: `npm install @mux-elements/[element_name]` or `yarn @mux-elements/[element_name]`.
  - (NOTE: If the issue is not occurring in the latest version, we may not resolve it.)
- [x] Confirmed that this issue hasn't [already been filed](https://github.com/muxinc/elements/issues?q=is%3Aopen+is%3Aissue+label%3Abug).

### Which Mux Elements/Packages has a bug?
- mux-video
<!--
  Uncomment all that apply. Also, make sure you add the appropriate Label after creating the issue!


- mux-audio
- common
-->

### Which browser(s) are you using?
- Chrome 103.0.5060.53
<!--
  Uncomment all that apply
  (NOTE: Currently, Mux Elements are only officially supported in new versions of the browsers listed above)


- Safari
- Firefox
- Edge ("Edgeium")
- Chrome Android
- Safari on iOS
- Other (list browser name(s), e.g. "Samsung Internet", "Opera"):
-->

### Which operating system(s) are you using?
- macOS Monterey 12.2

### How are you using Mux Elements?
- npm packages

### Describe the bug

- Not working with React 18.2
- Not working with --legacy-peer-deps either

<!--
  Please provide a clear and concise description of what the bug is. Include
  screenshots if needed. Please test using the latest version of the relevant
  Mux Elements packages to make sure your issue has not already been fixed.
-->

## Steps To Reproduce

1. Be on React 18.2
2. npm i @mux/mux-player-react
3.

<!--
  Your bug will get fixed much faster if we can run your code and it's clear how to reproduce the issue. Issues without reproduction steps or
  code examples may be immediately closed as not actionable.
-->

Link to code example:

<!--
  Please provide a CodeSandbox (https://codesandbox.io/s/new), a link to a
  repository on GitHub, or provide a minimal code example that reproduces the
  problem. You may provide a screenshot of the application if you think it is
  relevant to your bug report. Here are some tips for providing a minimal
  example: https://stackoverflow.com/help/mcve.
-->

## The current behavior

code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: ms-mel@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.0.2" from @mux/mux-player-react@0.1.0-canary.0-556dabf
npm ERR! node_modules/@mux/mux-player-react
npm ERR!   @mux/mux-player-react@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

## The expected behavior

--- a/packages/mux-audio-react/package.json
+++ b/packages/mux-audio-react/package.json
@@ -32,9 +32,9 @@
     "publish-release": "../../scripts/publish.sh"
   },
   "peerDependencies": {
-    "@types/react": "^16.8.6 || ^17.0.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "@types/react": "^17.0.0 || ^18",
+    "react": "^17.0.2 || ^18",
+    "react-dom": "^17.0.2 || ^18"
   },
   "peerDependenciesMeta": {
     "@types/react": {

--- a/packages/mux-player-react/package.json
+++ b/packages/mux-player-react/package.json
@@ -32,9 +32,9 @@
     "publish-release": "../../scripts/publish.sh"
   },
   "peerDependencies": {
-    "@types/react": "^16.8.6 || ^17.0.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "@types/react": "^17.0.0 || ^18",
+    "react": "^17.0.2 || ^18",
+    "react-dom": "^17.0.2 || ^18"
   },
   "peerDependenciesMeta": {
     "@types/react": {

--- a/packages/mux-video-react/package.json
+++ b/packages/mux-video-react/package.json
@@ -32,9 +32,9 @@
     "publish-release": "../../scripts/publish.sh"
   },
   "peerDependencies": {
-    "@types/react": "^16.8.6 || ^17.0.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "@types/react": "^17.0.0 || ^18",
+    "react": "^17.0.2 || ^18",
+    "react-dom": "^17.0.2 || ^18"
   },
   "peerDependenciesMeta": {
     "@types/react": {

